src/mod_offline_rdbms.erl

```diff
@@ -151,3 +153,27 @@ maybe_encode_timestamp(never) ->
     mongoose_rdbms:escape_null();
 maybe_encode_timestamp(TimeStamp) ->
     encode_timestamp(TimeStamp).
+
+get_personal_data(Username, Server) ->
+    LUser = jid:nodeprep(Username),
+    LServer = jid:nodeprep(Server),
+    SUser = mongoose_rdbms:escape_string(LUser),
+    SServer = mongoose_rdbms:escape_string(LServer),
+    TimeStamp = p1_time_compat:timestamp(),
+    STimeStamp = encode_timestamp(TimeStamp),
+    User = jid:to_binary({Username, LServer}),
+    {selected, Rows} = rdbms_queries:fetch_offline_messages(LServer, SUser, SServer, STimeStamp),
+    [{offline, ["timestamp", "from", "to", "packet"], rows_to_gdpr_format(User, Rows)}].
+
+
+rows_to_gdpr_format(User, Rows) ->
+    [row_to_gdpr_format(User, Row) || Row <- Rows].
+
+row_to_gdpr_format(User, {STimeStamp, SFrom, SPacket}) ->
+    Timestamp = usec:to_now(mongoose_rdbms:result_to_integer(STimeStamp)),
+    NowUniversal = calendar:now_to_universal_time(Timestamp),
+    {UTCTime, UTCDiff} = jlib:timestamp_to_iso(NowUniversal, utc),
+    UTC = list_to_binary(UTCTime ++ UTCDiff),
+    [UTC, jid:to_binary(jid:binary_to_bare(SFrom)), User, SPacket].
```

Comment:
JID is expected as third element, not username.

Reply:
actually it is JID (https://github.com/esl/MongooseIM/pull/2289/files#diff-4fe7d08b2813e3e1da98142afd6341b7R164). The variable name is confusing